includes/razorpay-subscriptions.php

```diff
@@ -124,10 +132,86 @@ protected function getSubscriptionCreateData($orderId)
         return $subscriptionData;
     }
 
-    protected function getProductPlanId($product, $order)
+    /**
+     * @param $subscriptionData
+     * @param $signUpFee
+     * @param $order
+     * @throws Errors\Error
+     */
+    protected function setStartAtIfNeeded(& $subscriptionData, & $signUpFee, $order)
     {
-        $currency = get_woocommerce_currency();
+        $product = $this->getProductFromOrder($order);
+
+        $sub = $this->getWooCommerceSubscriptionFromOrderId($order->get_id());
+
+        $metadata = get_post_meta($product['product_id']);
+
+        if (empty($metadata['razorpay_wc_start_date']) === false)
+        {
+            //
+            // When custom start date is set, we consider the initial payment
+            // as a sign up payment, and the first recurring payment will be
+            // made on the configured start date on the next month
+            //
+            $startDay = $metadata['razorpay_wc_start_date'][0];
+
+            $startDay = 40;
+
+            //
+            // $startDay must be in between 1 and 28
+            //
+            if (($startDay <= 1) or ($startDay >= 28))
```

Comment:
will always be true, since the above assignment isn't conditional

Reply:
@captn3m0 Yep. Posted above that this is a bug.